backends/rapidpro/backend.go

```diff
@@ -336,6 +341,71 @@ func (b *backend) WriteMsgStatus(ctx context.Context, status courier.MsgStatus)
 	return nil
 }
 
+// handleContactURNUpdate handles updating contact URN according to the old URN and the new URN from status
+func (b *backend) handleContactURNUpdate(ctx context.Context, status courier.MsgStatus) error {
+	// both URNs must have the same scheme
+	if status.OldURN().Scheme() != status.NewURN().Scheme() {
+		return errors.New("cannot update contact URN to a different URN scheme")
+	}
+	// ignore update if we have the same URN path
+	if status.OldURN().Path() == status.NewURN().Path() {
+		return nil
+	}
```

Comment:
These errors / checks can be in `SetUpdatedURN()` that way the error happens closer to where the mistake is made.

Reply:
But the logic will be duplicated to `DBMsgStatus` and `mockMsgStatus`.

Reply:
`mockMsgStatus` doesn't need these checks at all, that's just for testing. As I said, much better to throw errors closest to where they occur, this would all be delayed.